plugins/item_tasks/server/rest.py

```diff
@@ -34,17 +34,41 @@ def __init__(self):
     @autoDescribeRoute(
         Description('List all available tasks that can be executed.')
         .pagingParams(defaultSort='name')
+        .param('minFileInputs', 'Search tasks by minimum number of file inputs.', required=False,
+               dataType='int', default=0)
+        .param('maxFileInputs', 'Search tasks by maximum number of file inputs.', required=False,
+               dataType='int')
     )
     @filtermodel(model='item')
-    def listTasks(self, limit, offset, sort, params):
+    def listTasks(self, limit, offset, sort, minFileInputs, maxFileInputs, params):
         cursor = self.model('item').find({
             'meta.isItemTask': {'$exists': True}
         }, sort=sort)
 
+        if minFileInputs or maxFileInputs:
+            cursor = self._filterMinMaxFileInputs(cursor, minFileInputs, maxFileInputs)
+
         return list(self.model('item').filterResultsByPermission(
             cursor, self.getCurrentUser(), level=AccessType.READ, limit=limit, offset=offset,
             flags=constants.ACCESS_FLAG_EXECUTE_TASK))
 
+    def _filterMinMaxFileInputs(self, cursor, minFileInputs, maxFileInputs):
+        for item in cursor:
+            fileCount = 0
```

Comment:
How about:
```
fileCount = sum(
    input['type'] == 'file'
    for input in
    item['meta']['itemTaskSpec'].get('inputs', [])
)
```
This would be simpler, and would remove the need for `try...except`.